### ISSUE TITLE
deploy fails on circleCI

### ISSUE THREAD

**Issue by viz** [serverless/serverless#1941]
##### Serverless Framework Version: 
1.0 beta 2

##### Operating System: 
Ubuntu 14.04 (CircleCI build container)

##### Expected Behavior:
ran sls deploy --stage staging --region us-east-1

expected deploy to complete (works from local environment - MacOS El Capitan, Node 4.4.0, sls 1.0b2)

##### Actual Behavior:
seemed to hang on 'Updating Stack' and then eventually failed (presume Circle timed it out).
sometimes hung on 'Zipping Service' so not always exactly the same behaviour.

![lawpath_pusherservice__7_-_circleci](https://cloud.githubusercontent.com/assets/12402/17912590/e562a0fc-69d7-11e6-923e-4a781e0005f1.png)

**Comment by viz** [comment]
tested from CircleCI container using SSH and running deployment manually - deployed successfully.

Could this be related to the cli interacting with CircleCI? Not sure whether there is anything that can be done on the serverless side or not... I will raise it with Circle as well.

**Comment by tengis** [comment]
@viz It's more like your environment issue. Make sure your circleCI agent & ssh user both using same PATHs or environment variables

**Comment by viz** [comment]
OK, so resolved this by defining an npm script that runs the same command. No change to any environment variables or paths...

Not sure why this would make a difference - I'm sure someone here will understand the difference.

**Comment by viz** [comment]
had a similar problem locally where sls deploy (zipping plugin) was failing because it couldn't find /node_modules/.bin/babel even though I'd removed all dependencies on babel and removed the node_modules directories.

Resolved by deleting node_modules and re-installing.
![1__bash](https://cloud.githubusercontent.com/assets/12402/17919034/89912808-6a0d-11e6-9d0b-1208a6679d19.png)

rebuilding on CircleCI with cleared cache also resolved the problem there.

**Comment by franz-josef-kaiser** [comment]
@viz It would be easier to read if you would copy-paste that instead of a screenshot…

**Comment by viz** [comment]
@franz-josef-kaiser : no problem, I'll do so in future.

Update on this issue, we've determined that the size of the zip file was causing the problem as the CircleCI build container was running out of memory (during the upload we think).

The large zip file (~27MB) was the result of two issues:

1. we are using both serverless 0.5.x and 1.0 beta 2 for different projects. 0.5.x has previously been installed with the -g flag and so to prevent problems with existing projects, we have installed 1.0 beta 2 as a dev dependency of the individual services using 1.0.
2. the default sls deploy behaviour zips up all the installed node modules including dev dependencies, so the zip file contained serverless as well as all our testing modules .

To resolve this, we've now installed the serverless-webpack plugin that gives us more control over what gets bundled and deployed to AWS. As a result, the final zip file size was reduced to 2.3MB and we've had no problems with deploying from Circle since.
